### ISSUE TITLE
can i connect to specific schema using connconfig ?

### ISSUE THREAD

**Issue by saiteja2021** [jackc/pgx#1591]
Please describe the issue in detail. If this is a question about how to use pgx please use discussions instead.
	conn, err := pgxpool.ConnectConfig(ctx, config)
	if err != nil {
		return nil, nil, err
	}

**Comment by StevenACoffman** [comment]
Config is the configuration struct for creating a pool. It must be created by `ParseConfig` and then it can be
 modified. A manually initialized `ConnConfig` will cause `ConnectConfig` to panic. I construct the `connString` for ParseConfig using this:
```
type DBInfo struct {
	DBUser   string `mapstructure:"user,omitempty"`
	DBPass   string `mapstructure:"password,omitempty"`
	DBHost   string `mapstructure:"host,omitempty"`
	DBPort   string `mapstructure:"port,omitempty"`
	DBName   string `mapstructure:"dbname,omitempty"`
	DBSchema string `mapstructure:"search_path,omitempty"`
}

func NewDBInfo(
	user string,
	password string,
	host string,
	port string,
	dbName string,
	dbSchema string,
) *DBInfo {
	dbInfo := &DBInfo{
		DBUser:   user,
		DBPass:   password,
		DBHost:   host,
		DBPort:   port,
		DBName:   dbName,
		DBSchema: dbSchema,
	}
	return dbInfo
}

func (dbInfo *DBInfo) ConnectionString() string {
	x := map[string]string{}
	err := mapstructure.Decode(dbInfo, &x)
	if err != nil {
		return ""
	}

	pairs := []string{"sslmode=disable"}
	for k, v := range x {
		pairs = append(pairs, fmt.Sprint(k, "=", v))
	}
	return strings.Join(pairs, " ")
}
```

**Comment by jackc** [comment]
You can use the `RuntimeParams` attribute to set `search_path`.

https://github.com/jackc/pgx/blob/f59e8bf5551f403e6b7ec0912097bce85ea21351/pgconn/config.go#L42

**Comment by StevenACoffman** [comment]
Oh! Nice trick for when you want to change `search_path` after the fact.

**Comment by chr1shung** [comment]
just came across this issue, can I use the `RuntimeParams` to dynamically change the `search_path`(schema) which I'll query to?

**Comment by jackc** [comment]
@chr1shung `RuntimeParams` only affects the initial connection `search_path`. Once the connection is established you need to use `set`.

**Comment by chr1shung** [comment]
Can I use `Conn.Exec` to `SET` ? otherwise which other API should I use ?

**Comment by jackc** [comment]
> Can I use Conn.Exec to SET ?

Yes. Though remember that is only affects a single connection, not the whole pool.

**Comment by chr1shung** [comment]
Hi @jackc sorry for bothering you again

Let's just say my pgxpool has only one connection and if I Exec `SET search_path=foo` for the first query, the next time I acquires the connection the search_path is still on `foo`?

**Comment by jackc** [comment]
Usually, yes. But you shouldn't rely on that. Even if your pool is configured to only have one connection it is possible for the connection to be closed and another one created in between acquires.
